stat/combin/combin.go

```diff
@@ -284,6 +284,17 @@ func IndexToCombination(dst []int, idx, n, k int) []int {
 //  [ 1 2 0 ]
 // Cartesian panics if any of the provided lengths are less than 1.
 func Cartesian(lens []int) [][]int {
+	rows := NumCartesianProducts(lens)
+	out := make([][]int, rows)
+	for i := 0; i < rows; i++ {
+		out[i] = SubFor(nil, i, lens)
+	}
+	return out
+}
+
+// NumCartesianProducts calculates the number of products of a cartesian product set for the given lengths.
+// At least one length must be provided and all length values must be positive, otherwise this will panic.
+func NumCartesianProducts(lens []int) int {
```

Comment:
I think we can find a better name for this. I'm thinking `Card`. It's just the multiplier of all the dimensions, which is the cardinality. `Card` is really best matched with `IdxFor` and `SubFor`, so it's better to have a shorter name I think. That way it's something like 
```
dims := make([]int, 10)
for i := range dims {
     dims[i] = 3
}
card := combin.Card(dims)
```

Then the documentation is just "computes the cardinality of the multi-dimensional space whose dimensions have size specified by dims"

(and then change the input parameter to be `dims` to match `IdxFor`. I think at that point we can return 0 if len(dims) = 0, and panic if any are less than 1.

Reply:
Thanks for the suggestions.

The Naming has been applied in 148d721.

The second point, ```return 0 if len(dims)==0``` has been fixed in 53ff12d
Since that panic originally came from ```func Cartesian(lens []int) [][]int``` I added the panic there again to not change the behavior of that existing API. 3c15556 

I am not sure about the last one regarding keeping the existing behavior.